app.py

```diff
@@ -2,15 +2,35 @@
 import numpy as np
 from kmeans.parallel import ParallelKMeans
 from kmeans.sequential import SequentialKMeans
+import csv
 
 np.random.seed(1234)
 
 K = 3
-N = 10000
 M = 2
 max_iter = 10
 
-X = np.random.rand(N, M)
+def load_data(filename):
+    data = []
+    with open(filename, 'r') as csvfile:
+        reader = csv.reader(csvfile)
+        for row in reader:
+            data.append([float(x) for x in row])
+    return np.array(data)
+
+def count_csv_rows(csv_file):
+    row_count = 0
+    with open(csv_file, 'r') as file:
+        reader = csv.reader(file)
+        for row in reader:
+            row_count += 1
+    return row_count
```

Comment:
Might want to look into this
https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.count.html#pandas-dataframe-count